apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerParameter.cs

```diff
@@ -95,25 +129,23 @@ public override string ParameterName
         /// </summary>
         public SpannerDbType SpannerDbType
         {
-            get { return TypeCode.GetSpannerDbType(); }
-            set { TypeCode = value.GetSpannerTypeCode(); }
+            get;
+            set;
         }
 
         /// <inheritdoc />
         public override object Value
         {
-            get { return _value; }
+            get => _value;
             set
             {
-                if (TypeCode == TypeCode.Unspecified)
+                if (SpannerDbType.TypeCode == TypeCode.Unspecified && value != null)
```

Comment:
I see where you are coming from, but disagree.  Because of how these objects are used, its most educational to fail exactly on the problem parameter rather than deeply buried in a commitasync call.
serializing to json isn't a target scenario for spannerparameters.  The closest thing is designer serialization where this sort of thing is usually accomplished with an ISupportInitialize implementation.  I don't think this will be common at all, but willing to consider ISuportInitialize with another PR